src/java.desktop/unix/classes/sun/awt/X11/XDragSourceContextPeer.java

```diff
@@ -408,6 +445,10 @@ private static long findClientWindow(long window) {
             }
         }
 
+        if (isXWaylandDndAwareWindow(window)) {
```

Comment:
Yes, since it's a recursive function, I was just trying to understand if `isXWaylandDndAwareWindow `is called at right place
I tried a simple recursive program similar to `findClientWindow `which gives StackOverflowError so I was thinking if `findClientWindow  `can also give SOF and whether we should call `isXWaylandDndAwareWindow ` at l446 or l439

```
public class Recursion {
    public static void main(String[] args) {
        long value = 22;
        System.out.println(test(value));
    }

    private static long test(long value) {
        long arr[] = {1L,2L,3L};
        for (long i : arr) {
            long val = test(i);
            if (val != 0) {
                System.out.println("val " + val);
                return val;
            }
        }
        if (getValue(value) == 22) {
            return value;
        }
        return 0;
    }
    private static long getValue(long value) {
        return value;
    }
}
```

Reply:
Yes, this could happen if the implementation changes, currently the dummy window has no children.

Updated.